lesson2/README.md

```diff
@@ -1,14 +1,103 @@
+3. **Configuration files** are used by Android Studio to configure your app.
+
+You'll be writing most of your code in Java files.
+### 1. Java Files
+When you interact with an app on your phone, there are different "screens" that you can go to. Snapchat, for example has a "friends list" screen, a "picture-taking" screen, a "settings" screen, and so on.  The settings screen displays different things and needs completely different logic than the picture-taking screen.  Android separates these "screens" into different files called **activities** and **fragments**.
+
+#### [Activities](https://developer.android.com/guide/components/activities.html) (ex. MainActivity.java)
+
+Activities contain one or many fragments inside of them.  It handles the passing of information between fragments.  Most simple apps (like the ones you'll be making in the beginning of this class) will only need one activity.
+
+#### [Fragments](https://developer.android.com/guide/components/fragments.html) (ex. MainActivityFragment.java)
+Fragments are basically what we described as "screens" in our Snapchat example. Each fragment does a separate task and shows different components to the user. If you were making Snapchat, you might have a fragment for each swipeable screen. Fragments are useful for:
+1. Modularity - dividing complex activity code across fragments for better organization and maintenance.
+2. Reusability - placing behavior or UI parts into fragments that can be shared across multiple activities.
+3. Adaptability - representing sections of a UI as different fragments and utilizing different layouts depending on screen orientation and size.
+
+### 2. [Layout XML](https://developer.android.com/guide/topics/ui/declaring-layout.html) files (ex. fragment_main.xml)
+XML files are basically the CSS/HTML of the Android world. They define where each component goes in your activities and fragments. Each activity and fragment will have its own XML file that it can communicate with.
+
+### 3. Configuration Files
+#### [Manifest](https://developer.android.com/guide/topics/manifest/manifest-intro.html) (AndroidManifest.xml)
+The Manifest file defines settings about your app and what permissions it needs.  Whenever you download an app from the Play Store, it will show you what permissions the app needs (like internet, data storage, GPS location) so that users can be knowledgable about what their apps are using. These permissions are taken from the app's manifest file.
+
+#### [Gradle file](https://developer.android.com/studio/build/index.html) (build.grade (Module: app))
+Gradle is the thing that builds your app, like a compiler. Each time you launch your app to your phone to test it, gradle will run and compile your code.
+
+#### strings.xml
+Android coding conventions dictate that every user-facing string should be stored in this file instead of hardcoded.  This is so every string in your app can be easily translated in one file.
+
```

Comment:
idk, I didn't use colors.xml at all in MP.  And I thought it wasn't a layout file, so it might get confusing if it's under layout xml files.  and it's not a main important file.  Java files and layout xml files are where 90% of the code goes